app/src/main/java/pl/edu/zut/mad/appzut/fragments/WebPlanFragment.java

```diff
@@ -0,0 +1,148 @@
+package pl.edu.zut.mad.appzut.fragments;
+
+import android.annotation.SuppressLint;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.JavascriptInterface;
+import android.webkit.WebResourceResponse;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.Toast;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import pl.edu.zut.mad.appzut.R;
+import pl.edu.zut.mad.appzut.network.DataLoadingManager;
+import pl.edu.zut.mad.appzut.network.ScheduleEdzLoader;
+
+public class WebPlanFragment extends Fragment {
+
+    @BindView(R.id.web_view_plan)
+    WebView web;
+
+    @BindView(R.id.please_wait)
+    View pleaseWaitView;
+
+    public static final String USERNAME_ARG = "username";
+    public static final String PASSWORD_ARG = "password";
+
+    private String username;
+    private String password;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_web_plan, container, false);
+        ButterKnife.bind(this, view);
+        username = getArguments().getString(USERNAME_ARG);
+        password = getArguments().getString(PASSWORD_ARG);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        initWebView();
+    }
+
+    @SuppressLint("AddJavascriptInterface")
+    private void initWebView() {
+        pleaseWaitView.setVisibility(View.VISIBLE);
+        web.getSettings().setJavaScriptEnabled(true);
+        web.addJavascriptInterface(this, "android");
```

Comment:
`addJavascriptInterface` in old (<4.1) Android versions enabled security vulnerability. While we're loading only trusted pages, this might be flagged by antiviruses. Please check that (And if it turns out to be okay then `shouldOverrideUrlLoading` hack below may be converted to `@JavascriptInterface` as well, otherwise convert this to `shouldOverrideUrlLoading`)